src/MagnumPlugins/UfbxImporter/UfbxImporter.cpp

```diff
@@ -0,0 +1,1013 @@
+        SceneFieldData{SceneField::Transformation, nodeObjects, transformations, SceneFieldFlag::ImplicitMapping},
+        SceneFieldData{SceneField::Translation, nodeObjects, translations, SceneFieldFlag::ImplicitMapping},
+        SceneFieldData{SceneField::Rotation, nodeObjects, rotations, SceneFieldFlag::ImplicitMapping},
+        SceneFieldData{SceneField::Scaling, nodeObjects, scalings, SceneFieldFlag::ImplicitMapping}
+    });
+
+    /* All other fields have the mapping ordered (they get filed as we iterate
+       through objects) */
+    if(meshCount) arrayAppend(fields, {
+        SceneFieldData{SceneField::Mesh, meshMaterialObjects, meshes, SceneFieldFlag::OrderedMapping},
+        SceneFieldData{SceneField::MeshMaterial, meshMaterialObjects, meshMaterials, SceneFieldFlag::OrderedMapping},
+    });
+    if(lightCount) arrayAppend(fields, SceneFieldData{
+        SceneField::Light, lightObjects, lights, SceneFieldFlag::OrderedMapping
+    });
+    if(cameraCount) arrayAppend(fields, SceneFieldData{
+        SceneField::Camera, cameraObjects, cameras, SceneFieldFlag::OrderedMapping
+    });
+
+    /* Convert back to the default deleter to avoid dangling deleter function
+       pointer issues when unloading the plugin */
+    arrayShrink(fields, DefaultInit);
+
+    return SceneData{SceneMappingType::UnsignedInt, nodeCount, std::move(data), std::move(fields), scene};
+}
+
+const void* UfbxImporter::doImporterState() const {
+    return _state->scene.get();
+}
+
+UnsignedLong UfbxImporter::doObjectCount() const {
+    return _state->nodeCountWithSynthetic;
+}
+
+Long UfbxImporter::doObjectForName(const Containers::StringView name) {
+    ufbx_scene *scene = _state->scene.get();
+    ufbx_node *node = ufbx_find_node_len(scene, name.data(), name.size());
+    return node ? Long(node->typed_id) : -1;
+}
+
+Containers::String UfbxImporter::doObjectName(const UnsignedLong id) {
+    ufbx_scene *scene = _state->scene.get();
+    if(id < scene->nodes.count) {
+        return asString(scene->nodes[id]->name);
+    } else {
+        return {};
+    }
+}
+
+UnsignedInt UfbxImporter::doCameraCount() const {
+    return (UnsignedInt)_state->scene->cameras.count;
+}
+
+Int UfbxImporter::doCameraForName(const Containers::StringView name) {
+    return typedId(ufbx_find_element_len(_state->scene.get(), UFBX_ELEMENT_CAMERA, name.data(), name.size()));
+}
+
+Containers::String UfbxImporter::doCameraName(const UnsignedInt id) {
+    return asString(_state->scene->cameras[id]->name);
+}
+
+Containers::Optional<CameraData> UfbxImporter::doCamera(UnsignedInt id) {
+    ufbx_camera *cam = _state->scene->cameras[id];
+
+    if (cam->projection_mode == UFBX_PROJECTION_MODE_PERSPECTIVE) {
+        return CameraData{CameraType::Orthographic3D,
+            Vector2{(Float)cam->orthographic_size.x, (Float)cam->orthographic_size.y},
+            (Float)cam->near_plane, (Float)cam->far_plane,
```

Comment:
Oh, one style tip that might be useful -- I have `-Wold-style-cast` enabled, so all these will cause a warning. You can use a constructor cast such as

```suggestion
            Float(cam->near_plane), Float(cam->far_plane),
```

to avoid the warning in trivial numeric type casts. The potentially-dangerous cases that need extra atttention (such as pointer casts) should be a fully verbose `static_cast<>` or `reinterpret_cast<>` though.

For safely casting array views (including size recalculations) you may want to use [Containers::arrayCast()](https://doc.magnum.graphics/corrade/namespaceCorrade_1_1Containers.html#a8e6435208f2cda38505084a930a3e9ec).

Reply:
Yea I noticed the massive warning spam in CircleCI last night :D Just a holdout of me doing C for libraries, will change everything to appropriate C++ style casts!